web/templates/authorize.gohtml

```diff
@@ -0,0 +1,142 @@
+{{ define "body" }}
+  <h1 class="govuk-heading-xl">Mock GOV.UK One Login</h1>
+  <form method="post">
+    {{ if .Identity }}
+      <div class="govuk-form-group">
+        <fieldset class="govuk-fieldset">
+          <legend class="govuk-fieldset__legend">
+            Which user do you want to sign in with?
+          </legend>
+          <div class="govuk-radios" data-module="govuk-radios">
+            <div class="govuk-radios__item">
+              <input class="govuk-radios__input" id="user-1" name="user" type="radio" value="donor">
+              <label class="govuk-label govuk-radios__label" for="user-1">
+                Sam Smith (donor)
+              </label>
+            </div>
+            <div class="govuk-radios__item">
+              <input class="govuk-radios__input" id="user-2" name="user" type="radio" value="certificate-provider">
+              <label class="govuk-label govuk-radios__label" for="user-2">
+                Charlie Cooper (certificate provider)
+              </label>
+            </div>
+            <div class="govuk-radios__item">
+              <input class="govuk-radios__input" id="user-3" name="user" type="radio" value="random">
+              <label class="govuk-label govuk-radios__label" for="user-3">
+                Somebody Else (a random person)
+              </label>
+            </div>
+          </div>
+        </fieldset>
+      </div>
+    {{ else }}
+      {{ if .Sub }}
+        <div class="govuk-form-group">
+          <label class="govuk-label govuk-label--m" for="f-sub">OneLogin sub</label>
+          <div class="govuk-hint">Sign in using a OneLogin sub (to ignore, leave empty)</div>
+          <input class="govuk-input" type="text" name="sub" id="f-sub" />
+        </div>
+      {{ end }}
+
+      <div class="govuk-form-group">
```

Comment:
Further to my comment above we need this email AND the identity information provided by the other part of the if/else

Reply:
interesting, maybe OneLogin has changed a lot since our mock was first written, could you give me a link to your code to see what you are doing?

Reply:
https://github.com/ministryofjustice/opg-use-an-lpa/pull/2520/files#diff-6396026450e81aba5660a7c658efc67bbe6e3d84c11603b41114a8e0464ee530R37

This is the relevant function. We need emails to enable legacy account matching and migration. In any given development session we might use a number of emails and so having it hardcoded (or supplied via env var) slows things down quite a lot.

Although we verify identity information validity we don't actually use anything thats provided.

Reply:
I think I've found it https://github.com/ministryofjustice/opg-use-an-lpa/blob/main/service-api/app/src/App/src/Service/Authentication/OneLoginService.php#L45. I thought you were only going to be doing authentication, but this param triggers the identity flow?

Reply:
As per https://docs.sign-in.service.gov.uk/integrate-with-integration-environment/authenticate-your-user/#make-a-request-to-the-authorize-endpoint this should be doing an authentication request and not care about identity (if it did need identity then that should be a separate request with an already signed in session, as modernise does). Then the current implementation will work fine

Reply:
Erm, we do do an authentication request. We also verify identity.

You've just broken this image for use's purpose. Gee thanks.